src/router/index.ts

```diff
@@ -14,4 +18,21 @@ const router = new VueRouter({
   }
 })
 
+router.beforeEach((to, from, next) => {
+  // If the route requires authentication
+  if (to.matched.some(record => record.meta.requiresAuth)) {
+    // If we  dont have a token
+    if (!sessionStorage.getItem(SessionStorageKeys.KeyCloakToken)) {
+      // Send them to the bcsc login
+      return next({
+        path: `/signin/bcsc${to.path}`,
+        query: { redirect: to.fullPath }
```

Comment:
If this line of code has the effect of passing the current route (eg, define-company) to the signin component, then the signin component can route back to it after successful login.

Reply:
(So that the signin component is no longer hard-coded to redirect to the define-business route.)

Reply:
There's some question how this works and whether my proposed change will handle all use cases, so let's go with the existing code for now and change later if needed.